### PR TITLE
Add sessionId parameters for create async query API

### PR DESCRIPTION
**Pull request by penghuo**
### Description
1. API doc. https://github.com/opensearch-project/sql/blob/2b1ad6a912a91f64056446aa24601ee1f1ab469c/docs/user/interfaces/asyncqueryinterface.rst
2. feature disable by default.

#### Todo
1. create .query_execution_request index if not exist.
3. change REPL job Spark parameters.
4. add IT in following PR.
5. enable the feature by default.

#### Test Queries
* enable feature.
```
PUT {{baseUrl}}/_plugins/_query/settings
Content-Type: application/x-ndjson

{"transient":{"plugins.query.executionengine.spark.session.enabled":"true"}}

{
  "acknowledged": true,
  "persistent": {},
  "transient": {
    "plugins": {
      "query": {
        "executionengine": {
          "spark": {
            "session": {
              "enabled": "true"
            }
          }
        }
      }
    }
  }
}
```
* submit query without sessionId, new sessionId will be allocated.
```
POST {{baseUrl}}/_plugins/_async_query
Content-Type: application/x-ndjson

{
  "datasource": "mys3",
  "lang": "sql",
  "query": "select * from mys3.default.http_logs limit 1"
}

{
  "queryId": "HlbM61kX6MDkAktO",
  "sessionId": "1Giy65ZnzNlmsPAm"
}
```
* fetch query result
```
GET {{baseUrl}}/_plugins/_async_query/HlbM61kX6MDkAktO

{
  "status": "waiting"
}
```
* submit new query in same session
```
POST {{baseUrl}}/_plugins/_async_query
Content-Type: application/x-ndjson

{
  "datasource": "mys3",
  "lang": "sql",
  "query": "select * from mys3.default.http_logs limit 1",
  "sessionId" : "1Giy65ZnzNlmsPAm"
}

{
  "queryId": "7GC4mHhftiTejvxN",
  "sessionId": "1Giy65ZnzNlmsPAm"
}
```
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementId.java [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class StatementId {
   private final String id;
 
   public static StatementId newStatementId() {
-    return new StatementId(RandomStringUtils.random(10, true, true));
+    return new StatementId(RandomStringUtils.random(16, true, true));
```
Comment (penghuo) [inline spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementId.java:16]: change length from 10 to 16, making it similar to EMR-S job id.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java [hunk 2]
```diff
@@ -111,23 +122,60 @@ public JSONObject getQueryResponse(AsyncQueryJobMetadata asyncQueryJobMetadata)
...
       result.put(ERROR_FIELD, error);
     } else {
-      // make call to EMR Serverless when related result index documents are not available
-      GetJobRunResult getJobRunResult =
-          emrServerlessClient.getJobRunResult(
-              asyncQueryJobMetadata.getApplicationId(), asyncQueryJobMetadata.getJobId());
-      String jobState = getJobRunResult.getJobRun().getState();
-      result.put(STATUS_FIELD, jobState);
-      result.put(ERROR_FIELD, "");
+      if (asyncQueryJobMetadata.getSessionId() != null) {
```
Comment (dai-chen) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java:125]: so this means the feature must be enabled at the moment?

Comment (penghuo) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java:125]: yes. later, we can decide whether we want to enable by default.

**Review by kaituo — APPROVED** [review]

File: opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java [hunk 3]
```diff
@@ -135,6 +135,13 @@ public class OpenSearchSettings extends Settings {
           Setting.Property.NodeScope,
           Setting.Property.Dynamic);
 
+  public static final Setting<?> SPARK_EXECUTION_SESSION_ENABLED_SETTING =
+      Setting.boolSetting(
+          Key.SPARK_EXECUTION_SESSION_ENABLED.getKeyValue(),
```
Comment (vamsi-amazon) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java:140]: Will we make this to True in future?

Comment (vamsi-amazon) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java:140]: Got my answer below.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java [hunk 4]
```diff
@@ -198,8 +246,35 @@ private DispatchQueryResponse handleNonIndexQuery(DispatchQueryRequest dispatchQ
             tags,
             false,
             dataSourceMetadata.getResultIndex());
-    String jobId = emrServerlessClient.startJobRun(startJobRequest);
-    return new DispatchQueryResponse(jobId, false, dataSourceMetadata.getResultIndex());
+    if (sessionManager.isEnabled()) {
```
Comment (vamsi-amazon) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java:249]: Why only in handleNonIndexQuery?

File: spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementId.java (hunk 1 above)
Comment (vamsi-amazon) [reply spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementId.java:16]: Nit::  randomAlphaNumeric

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java (hunk 4 above)
Comment (vamsi-amazon) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java:249]: Got it Index Query doesn't need session
and ppl is handled via handleNonIndexQuery.

**Review by vamsi-amazon — DISMISSED** [review]

File: spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementId.java (hunk 1 above)
Comment (penghuo) [reply spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementId.java]: done.

File: docs/user/admin/settings.rst [hunk 5]
```diff
@@ -311,3 +311,39 @@ SQL query::
...
+1. The default value is false.
+2. This setting is node scope.
+3. This setting can be updated dynamically.
+
+You can update the setting with a new value like this.
+
+SQL query::
+
+    sh$ curl -sS -H 'Content-Type: application/json' -X PUT localhost:9200/_plugins/_query/settings \
+    ... -d '{"transient":{"plugins.query.executionengine.spark.session.enabled":"false"}}'
```
[the remark this answers is not in the document]
Comment (penghuo) [reply docs/user/admin/settings.rst]: done.

**Review by dai-chen — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Review by anirudha — APPROVED** [review]

**Pull request merged**